### PR TITLE
add stable-code-3b local model support

### PR DESCRIPTION
**Pull request by xixingya**

### CODE REVIEW

**Comment by xixingya** [comment]
add auth before v1 and add stable-code-3b local model support

File: main.go [hunk 1]
```diff
@@ -317,11 +362,12 @@ func main() {
 		return
 	}
 
-	proxyService.InitRoutes(r)
+	proxyService.InitRoutes(r, cfg)
```
Comment (wozulong) [inline main.go:365]: InitRoutes 中可以通过 s.cfg 拿到 cfg

File: main.go [hunk 2]
```diff
@@ -256,11 +283,29 @@ func (s *ProxyService) codeCompletions(c *gin.Context) {
 
 	body, _ = sjson.DeleteBytes(body, "extra")
 	body, _ = sjson.DeleteBytes(body, "nwo")
-	body, _ = sjson.SetBytes(body, "model", InstructModel)
-
+	suffix := gjson.GetBytes(body, "suffix")
+	prompt := gjson.GetBytes(body, "prompt")
+	content := fmt.Sprintf("<fim_prefix>%s<fim_suffix>%s<fim_middle>", prompt, suffix)
```
Comment (wozulong) [inline main.go:288]: 这里 chat completions 和单纯的 completions 是不是可以模型区分一下，最好拆出两个方法出来呢？

Comment (xixingya) [reply main.go]: fixed PTAL

**Comment by xixingya** [comment]
@wozulong  PTAL

**Comment by xixingya** [comment]
I have tested, gpt-3.5-instruct and stable code can work normally.

**Comment by wozulong** [comment]
Thank you!

**Pull request merged**